v-next/hardhat/src/internal/builtin-plugins/solidity/build-system/cache.ts

```diff
@@ -0,0 +1,74 @@
+import path from "node:path";
+
+import {
+  exists,
+  getAccessTime,
+  getAllFilesMatching,
+  getSize,
+  readJsonFile,
+  remove,
+  writeJsonFile,
+} from "@ignored/hardhat-vnext-utils/fs";
+
+export class ObjectCache<T> {
+  readonly #path: string;
+  readonly #defaultMaxAgeMs: number;
+  readonly #defaultMaxSize: number;
+
+  constructor(
+    basePath: string,
+    namespace: string,
+    version: string,
+    defaultMaxAgeMs: number = 7 * 24 * 60 * 60 * 1000, // 1 week
+    defaultMaxSize: number = 2 * 1024 * 1024 * 1024, // 2 GB
+  ) {
+    this.#path = path.join(basePath, namespace, version);
+    this.#defaultMaxAgeMs = defaultMaxAgeMs;
+    this.#defaultMaxSize = defaultMaxSize;
+  }
+
+  public async set(key: string, value: T): Promise<void> {
+    const filePath = path.join(this.#path, key);
+    await writeJsonFile(filePath, value);
+  }
+
+  public async get(key: string): Promise<T | undefined> {
+    const filePath = path.join(this.#path, key);
+    return (await exists(filePath)) ? readJsonFile<T>(filePath) : undefined;
+  }
+
+  public async clean(maxAgeMs?: number, maxSize?: number): Promise<void> {
```

Comment:
Being able to override the default max age and size is really useful for testing purposes.